### ISSUE TITLE
Question: "local" vs "default" zone

### ISSUE THREAD

**Issue by balagge** [moment/luxon#558]
I am new to luxon, and there is some confusion that I cannot really sort out. Maybe it is a documentation issue, or just something that is not clear to me.

Based on the definition of "local" and "default" timezone in the docs, I assumed that
- "local" always means the local zone (where the code executes), and
- "default" is the zone that is applied if there is no zone explicitly indicated on an operation

These definitions are different (to a new user like me)

Example:
```javascript
Settings.defaultZoneName = "utc";
console.log(DateTime.local());
console.log(DateTime.fromObject({zone:"local"}));
```

Contrary to what I would expect, both objects created are in UTC.

Some searching led me to #333, which seems to imply that this is the intended behavior, which seems odd, because it makes the `{zone:"local"}` setting identical to no `zone` specified at all, so the `{zone:"local"}` as well as the method `DateTime.local()` seems to be misleadingly named. 

It would make sense to rename those to `{zone:"default"}` and `DateTime.defaultZone()` or similar. Alternatively, the `Settings` object should have "localZone" instead of "defaultZone" properties.

The bottom line is that the concept of "default zone" seems to be the same as "local zone". Or is there a difference I do not get?

**Comment by GillesDebunne** [comment]
Hello,

When using `"local"` to define a Zone, the `Settings`' default zone is always used. Your code is hence working as expected.

The Settings' default zone is indeed the system's *local zone* by default, which explains the name. 

However, for tests environments or other specific purposes, it might be useful to be able to override the default zone, in order to pretend that the application is running in a specific time zone. When this is done, *all* constructors use this Zone, unless otherwise specified. The `local` keyword and the `local()` method then also refer to this overridden zone.

This behavior might be confusing, and in version 2.x we are deprecating the `local` keyword in favor of `default`. This hopefully makes it clearer that unless specified, each DateTime constructor will use the Setting's `defaultZone` (which is still the system's local zone unless defined otherwise).

We also introduce the new `system` keyword for zones, which always refers to the system's time zone, irrespective of the Settings' `defaultZone`. See #469 for discussion.

Hope this makes it clearer, and you like the coming changes.

**Comment by balagge** [comment]
Thank you for the explanation.
I like the proposed solution ( `system` / `default` instead of `local`) very much.
